lib/utils.py

```diff
@@ -46,3 +53,28 @@ def is_in_uk(simple_polygons):
         first_coordinate[1] > uk_south_west[1] and
         first_coordinate[1] < uk_north_east[1]
     )
+
+def upload_to_s3(rendered_pages):
+    s3 = boto3.resource('s3')
+    bucket_name = os.environ['GOVUK_ALERTS_BUCKET_NAME']
+
+    for path, content in rendered_pages.items():
+        logger.info("Uploading " + path)
+        item = s3.Object(bucket_name, path)
+        item.put(Body=content, ContentType="text/html")
```

Comment:
Yes it is expected: https://github.com/alphagov/notifications-broadcasts-infra/blob/main/terraform/modules/govuk-alerts-iam/site.tf#L44-L47

Reply:
Is it worth temporarily removing the restriction in Staging so we can test it out? I expect there's at least one bug we haven't spotted together, and it's just easier to fix things without having to deploy them first.

Reply:
I did and raised alphagov/notifications-aws#938 to fix the issues

Reply:
Nice. I'll try and take a look today.

Related to my initial comment about the sync behaviour changing, I don't think we need to do anything in this PR. It might surprise someone in future, but I can't think of a sure way to avoid that. I'll resolve this one.

Reply:
Ah, yes, about the sync behaviour, I think it's not _as_ straight forward to do it when the objects and their content comes from an in-memory object instead of files on disk.

I think in the future we can implement this ourselves by listing all the non-asset keys in the bucket, getting the difference between them and what we're about to upload and, after we upload and clean the cache we can go ahead and remove all the remaining ones.